api/hardwaremanagement/v1alpha1/node_pools.go

```diff
@@ -57,8 +57,8 @@ type NodeGroup struct {
 	Size      int    `json:"size" yaml:"size"`
 	// +kubebuilder:validation:MinItems=1
 	Interfaces     []string `json:"interfaces,omitempty"`
-	ResourcePoolId string   `json:"resourcePoolId"`
-	Role           string   `json:"role"`
+	ResourcePoolId string   `json:"resourcePoolId,omitempty"`
+	Role           string   `json:"role,omitempty"`
```

Comment:
The new fields are populated from the hardware template. However, I don't see explicit code to populate these fields during rendering of the hardware template.

Reply:
The new `ResourcePoolId` and `Role` fields would get populated automatically, due to the unmarshalling of the json when getting the NodeGroup from the hwtemplate, right? So it was just the `Extensions` processing that was missing?

Reply:
The user inputs are sourced from the hardware template ConfigMap's data fields, which are currently free-form. At this time, individual field assignments are necessary to ensure the functionality works as expected.

Reply:
These aren't fields directly under spec, though, they're part of the NodeGroup struct and would be populated as part of the nodegroup

Reply:
Right, I missed it.